### DISCUSSION TITLE
Support for RTL languages , specially Arabic

### DISCUSSION THREAD

**Discussion by TheWondaBoi** [Chatterino/chatterino2#3805] [Q&A] [answered]
hello 
i know this has been discussed but i've never understood what was the outcome of that discussion 

i mean is it possible to have RTL languages supported in chatterino , is there any workaround for us users?

please i need simple explaination as i know that this is duplicated but i never understood the outcome of the older threads about this subject

**Comment by leon-richardt** [comment]
**TL;DR:** It's generally recognized as a shortcoming of Chatterino that RTL languages are not supported. Unfortunately, implementing support for them is not straightforward and would probably require a significant amount of rewriting core layouting logic at a fairly low level in Chatterino. Making sure nothing breaks as a result of this rewrite is not trivial and will probably also take a good chunk of work.

Feel free to let me know if you have further questions.

--------

Relevant issues for reference:
- https://github.com/Chatterino/chatterino2/issues/720
- https://github.com/Chatterino/chatterino2/pull/1970
- https://github.com/Chatterino/chatterino2/issues/930
- https://github.com/Chatterino/chatterino2/discussions/3794
- ... and probably others I have missed.

**Comment by Felanbird** [comment, marked as answer]
@TheWondaBoi Proper RTL support was added in #3958 - I attempted to @ everybody who ever asked about this feature, in #720, but it appears I missed those who opened discussions.
